prusti-viper/src/encoder/type_encoder.rs

```diff
@@ -392,6 +405,28 @@ impl<'p, 'v, 'r: 'v, 'tcx: 'v> TypeEncoder<'p, 'v, 'tcx> {
                 }
             }
 
+            ty::TyKind::Array(elem_ty, len) => {
+                let array_len =
+                    self.encoder
+                        .const_eval_intlike(&len.val).unwrap()
+                        .to_u64().unwrap();
+                // TODO: use snapshot type at some point
+                // let snapshot = self.encoder.encode_snapshot(&self.ty)?;
+                // let return_ty = snapshot.get_type();
+
+                // need the array's corresponding lookup_pure function as well
+                self.encoder.encode_builtin_function_def(
+                    BuiltinFunctionKind::ArrayLookupPure {
+                        array_ty: vir::Type::TypedRef(format!("{}", elem_ty)),
+                        array_len: array_len.try_into().unwrap(),
+                        return_ty: vir::Type::Int,
+                    });
```

Comment:
For the time being, this is fine, but eventually the encoding of `lookup_pure` should be triggered by actually using it where it is needed.

Reply:
also should be called lazily by `encode_builtin_function_use` at the top of the function, as [@fpoli noted](https://github.com/viperproject/prusti-dev/pull/463/files/70542500f5013c218a5332fe93f58e79c2ee135f#r614717183). will remove, anyway.

Reply:
wait, that's `encode_type_predicate_use` on top there, not `encode_builtin_function_use`. the latter is only called from `pure_function_encoder.rs`, so i think i want to have it called somewhere "manually" for now after all. i'll add a TODO to remove it once we have the `encode_builtin_function_use` calls from other places.